pyteal/ast/acct_test.py

```diff
@@ -1,79 +1,63 @@
+        ("totalAppsOptedIn", "total_apps_opted_in"),
+        ("totalAssetsCreated", "total_assets_created"),
+        ("totalAssets", "total_assets"),
+        ("totalBoxes", "total_boxes"),
+        ("totalBoxBytes", "total_box_bytes"),
+    ],
+)
+def test_acct_param_fields_valid(method_name, field_name):
     arg = pt.Int(1)
-    expr = pt.AccountParam.balance(arg)
+    account_param_method = getattr(pt.AccountParam, method_name)
+    expr = account_param_method(arg)
     assert expr.type_of() == pt.TealType.none
-    assert expr.value().type_of() == pt.TealType.uint64
+
+    account_param_field = AccountParamField[field_name]
+    assert expr.value().type_of() == account_param_field.type_of()
 
     expected = pt.TealSimpleBlock(
         [
             pt.TealOp(arg, pt.Op.int, 1),
-            pt.TealOp(expr, pt.Op.acct_params_get, "AcctBalance"),
+            pt.TealOp(expr, pt.Op.acct_params_get, account_param_field.arg_name),
             pt.TealOp(None, pt.Op.store, expr.slotOk),
             pt.TealOp(None, pt.Op.store, expr.slotValue),
         ]
     )
 
-    actual, _ = expr.__teal__(avm6Options)
-    actual.addIncoming()
-    actual = pt.TealBlock.NormalizeBlocks(actual)
-
-    with pt.TealComponent.Context.ignoreExprEquality():
-        assert actual == expected
-
-
-def test_acct_param_min_balance_valid():
-    arg = pt.Int(0)
-    expr = pt.AccountParam.minBalance(arg)
-    assert expr.type_of() == pt.TealType.none
-    assert expr.value().type_of() == pt.TealType.uint64
-
-    expected = pt.TealSimpleBlock(
-        [
-            pt.TealOp(arg, pt.Op.int, 0),
-            pt.TealOp(expr, pt.Op.acct_params_get, "AcctMinBalance"),
-            pt.TealOp(None, pt.Op.store, expr.slotOk),
-            pt.TealOp(None, pt.Op.store, expr.slotValue),
-        ]
+    supported_options_version = pt.CompileOptions(
+        version=account_param_field.min_version
     )
-
-    actual, _ = expr.__teal__(avm6Options)
+    actual, _ = expr.__teal__(supported_options_version)
     actual.addIncoming()
     actual = pt.TealBlock.NormalizeBlocks(actual)
 
     with pt.TealComponent.Context.ignoreExprEquality():
         assert actual == expected
 
-
-def test_acct_param_auth_addr_valid():
-    arg = pt.Int(1)
-    expr = pt.AccountParam.authAddr(arg)
-    assert expr.type_of() == pt.TealType.none
-    assert expr.value().type_of() == pt.TealType.bytes
-
-    expected = pt.TealSimpleBlock(
-        [
-            pt.TealOp(arg, pt.Op.int, 1),
-            pt.TealOp(expr, pt.Op.acct_params_get, "AcctAuthAddr"),
-            pt.TealOp(None, pt.Op.store, expr.slotOk),
-            pt.TealOp(None, pt.Op.store, expr.slotValue),
-        ]
-    )
-
-    actual, _ = expr.__teal__(avm6Options)
-    actual.addIncoming()
-    actual = pt.TealBlock.NormalizeBlocks(actual)
-
-    with pt.TealComponent.Context.ignoreExprEquality():
-        assert actual == expected
+    with pytest.raises(pt.TealInputError):
```

Comment:
@jdtzmn Since we're discussing the `verifyProgramVersion` check in another thread, I'm wondering if you're up for extending the highlighted test (or another one) to validate the check happens for `acct_params_get`?

I'm not tied to the implementation, but as an example:
```
    def test_unsupported_version(version: int):
        with pytest.raises(pt.TealInputError):
            unsupported_options_version = pt.CompileOptions(
                version= version
            )
            expr.__teal__(unsupported_options_version)
    test_unsupported_version(account_param_field.min_version - 1)
    test_unsupported_version(pt.ir.Op.acct_params_get.min_version - 1)
```

Reply:
Yes, good idea!